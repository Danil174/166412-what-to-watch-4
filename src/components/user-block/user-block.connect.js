import {connect} from "react-redux";
import {getAuthorizationStatus, getUserData} from "../../reducer/user/selectors.js";
import UserBlock from "./user-block.jsx";

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  userData: getUserData(state),
});

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
