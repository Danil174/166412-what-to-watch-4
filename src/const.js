export const TextMovieRatings = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const SmallCardVideoSettings = {
  WIDTH: `100%`,
  HEIGHT: `100%`,
  IS_MUTED: true,
};

export const DEFAULT_GENRE = `All genres`;
export const MAX_GENRES_LENGTH = 9;
export const INITTIAL_LIST_LENGTH = 8;
export const SIMILAR_LIST_LENGTH = 4;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const smallCardVideoStartTimeOut = 1000;

export const LoadErrorsTexts = {
  LIST_FAIL: `Не удалось загрузить список фильмов.`,
  PROMO_FAIL: `Не удалось загрузить промо фильм.`,
};

export const AppRoute = {
  LOGIN: `/login`,
  MOVIE_PAGE: `/movie_page`,
  FAVORITES: `/favorites`,
  ROOT: `/`,
};
