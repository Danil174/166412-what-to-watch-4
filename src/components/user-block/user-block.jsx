import React from "react";
import PropTypes from "prop-types";

import {AuthorizationStatus, AppRoute} from "../../const.js";

import Avatar from "../avatar/avatar.jsx";

const SignIn = () => {
  return (
    <a href={AppRoute.LOGIN} className="user-block__link">Sign in</a>
  );
};

const UserBlock = (props) => {
  const {authorizationStatus, userData} = props;
  const {avatarUrl} = userData;

  return (
    authorizationStatus === AuthorizationStatus.AUTH
      ? <Avatar url={avatarUrl} /> : <SignIn />
  );
};

UserBlock.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  userData: PropTypes.shape({
    avatarUrl: PropTypes.string,
  }),
};

export default UserBlock;
