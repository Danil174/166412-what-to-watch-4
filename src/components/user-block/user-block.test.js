import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import {AuthorizationStatus} from "../../const.js";

import {UserBlock} from "./user-block.jsx";

const mockStore = configureStore([]);

describe(`UserBlock`, () => {
  it(`render UserBlock`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationStatus: AuthorizationStatus.NO_AUTH,
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <UserBlock
              authorizationStatus={`AUTH`}
            />
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
