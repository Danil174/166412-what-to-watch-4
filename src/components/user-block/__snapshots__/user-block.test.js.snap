// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock render UserBlock 1`] = `
<div
  className="user-block__avatar"
>
  <img
    alt="User avatar"
    height="63"
    src="img/avatar.jpg"
    width="63"
  />
</div>
`;
